frontend/src/app/companies/profile/company-profile/company-profile.component.ts

```diff
@@ -0,0 +1,34 @@
+import { CommonModule } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+import { provideComponentStore } from '@ngrx/component-store';
+
+import { ContentErrorComponent } from '@app/ui/content-error/content-error.component';
+import { ContentLoadingComponent } from '@app/ui/content-loading/content-loading.component';
+
+import { CompanyProfileStore } from './company-profile.store';
+
+@Component({
+    selector: 'app-company-profile',
+    standalone: true,
+    imports: [CommonModule, ContentErrorComponent, ContentLoadingComponent],
+    providers: [provideComponentStore(CompanyProfileStore)],
+    templateUrl: './company-profile.component.html'
+})
+export class CompanyProfileComponent implements OnInit {
+    id?: number;
+    readonly vm$ = this.companyProfileStore.vm$;
+
+    constructor(
+        private readonly companyProfileStore: CompanyProfileStore,
+        private route: ActivatedRoute
+    ) {}
+
+    ngOnInit(): void {
+        this.id = Number(this.route.snapshot.paramMap.get('id'));
+        if (this.id) {
```

Comment:
minor: I would just add a validation in case we don't find the id in the db, to redirect the user to the companies list page.

Reply:
Not sure if I chose the best way to do it, let me know if there's a better way

Reply:
it's good for now, we'll need to use the store and check for a 404 later, but we don't have to do it right now, we can do it in the end if we have time for it - thanks!